### DISCUSSION TITLE
Propagate PEP 258 Attribute Docstrings to dataclass constructor and TypedDict keys

### DISCUSSION THREAD

**Discussion by multimeric** [microsoft/pylance-release#3120] [Enhancement]
## Dataclasses
Currently, we can use the PEP 258 attribute docstring syntax to document dataclass fields, and it correctly shows up when we try to *access* these fields:
![image](https://user-images.githubusercontent.com/5019367/182167703-4c528a8a-823c-48f0-ab07-ee3dbecf45f3.png)

However when we try to construct such an object, we don't get the same parameter description, even though it refers to the same field, so could have the same description:
![image](https://user-images.githubusercontent.com/5019367/182167613-ea4af445-90e1-4144-95e4-5091877734e7.png)

## TypedDict

Similarly, `TypedDict` uses the class syntax, so we can use the docstring syntax again. And again, it doesn't show up in the constructor, or in a dictionary literal with the appropriate type:

![image](https://user-images.githubusercontent.com/5019367/182166805-13c2293f-4ca5-446d-bafb-5a19c2c0fc05.png)
![image](https://user-images.githubusercontent.com/5019367/182167132-12e0ad67-f938-459d-8509-18484d9f100c.png)

As with dataclasses, it would be very helpful to users for this information to be propagated to the constructor and/or dictionary literal.

**Comment by multimeric** [comment]
Okay so it seems that if you trigger "Show Definition Preview Hover" or just "Show Hover" (they're different commands), then you do get something useful. And it works with both `TypedDict` and dataclasses too! It seems that by default "Show Hover" is bound to Ctrl+K, Ctrl+I, which gives you something like this:

![image](https://user-images.githubusercontent.com/5019367/182181593-a6af9ed7-c7b4-47ef-8eef-073ba5749316.png)

![image](https://user-images.githubusercontent.com/5019367/182181892-8fd7c951-8659-4b12-9232-ce5369bcba83.png)

**Comment by benglewis** [comment]
Just to emphasize, as per #4759 , it does not actually show the docstring inside the main class constructor call, but instead, on the field-by-field level, meaning you must first choose to write a keyword argument to see the docstring for those fields. It would be nice if Pylance could add the docstring to the class docstring too if there is not already one there
